internal/controller/result.go

```diff
@@ -38,11 +42,43 @@ func RequeueIfUnavailable(err error) (*ctrl.Result, error) {
 
 // RequeueConflict will return a new result and the supplied error, adjusted for Kubernetes conflict errors
 func RequeueConflict(err error) (*ctrl.Result, error) {
+	return requeueConflictForFrameStack(err, getFrames())
+}
+
+func requeueConflictForFrameStack(err error, fs frameStack) (*ctrl.Result, error) {
 	result := &ctrl.Result{}
 	if apierrs.IsConflict(err) {
-		ReconcileConflictErrors.WithLabelValues(guessController()).Inc()
+		controllerName := guessController(fs)
+		ReconcileConflictErrors.WithLabelValues(controllerName).Inc()
 		result.Requeue = true
 		err = nil
 	}
 	return result, err
 }
+
+// guessController dumps stack to try and guess what the controller name should be
```

Comment:
This was a hack when I wanted to get metrics in so we could keep track of where we were seeing conflicts. In hindsight it would probably be better if we did something similar to what we do with logs; that is have a `Metrics` interface with a method like `ReconcileConflictErrors() prometheus.Counter` and store and instance with the populated labels in a field on relevant `*Reconciler` types (populated in `main.go`). We could hang the labelled metrics instance off the context at the start of the `Reconcile` method and change the signature above to `RequeueConflict(ctx context.Context, err error)...`.